atef/widgets/config.py

```diff
@@ -185,6 +185,8 @@ def save_as(self, *args, filename: Optional[str] = None, **kwargs):
                 caption='Save as',
                 filter='Json Files (*.json)',
             )
+        if filename.endswith('.json') is False:   # json_extension_bug_fix location
```

Comment:
Seems reasonable to me. We might still have some weird cases with someone intentionally giving themselves a different extension and getting surprised here, but considering the gui is looking for .json files it's probably OK to force it.

Let's clean up just a little bit: we can remove the comment (the PR/commits are enough documentation) and we don't need to check if something `is False`. The normal python way to do the check is simply:
```suggestion
        if not filename.endswith('.json'):
```